### ISSUE TITLE
rank up & leveling tool questions/issues

### ISSUE THREAD

**Issue by drchia** [shalzuth/BraveHaxvius#36]
First off, thanks for the amazing work with this tool. Hell of a job and well worth the donation!

With regards to the rank up feature, it now stops once it detects you are at rank 150, which I'm assuming is due to it being the cap until just recently. Can it be updated so that it continues until 200?

Second, and this is more of a question than an issue, does the level up feature still attempt to equip eggs/action since injection no longer works? I had a problem recently with the older version where the tool sold all of my eggs and action materia because it thought it had injected them. No complaints of course as I simply needed to re-farm them, but I was hesitant about using the feature again if the same thing may happen. I know the new version said the leveling tool was updated, but I wasn't sure if this is what it meant.

Thanks!

**Comment by drchia** [comment]
Sorry, disregard the second question re: level up feature. I see in the code that the egg/action equip has been removed within UnitHunterNew. Also, I see that the rank tool does indeed break at rank 150, and only at 150 (it works for 151+ and will actually keep running even if you are 200). Unfortunately, while I'm competent enough to read through code and locate easy to spot things, such as the above, I'm nowhere near proficient enough to actually change that code within an exe that I downloaded lol.

**Comment by WithinGDR** [comment]
Get Visual Studio and just change 150 to 200

**Comment by drchia** [comment]
Unfortunately, I'm not a programmer by any means so I have absolutely no clue how to use visual studio to edit anything. I can simply follow easy to understand code to find certain things. It's not hard for someone to see that there's an "if" statement that says "if rank=150" tied to the rank up button. I just have no idea how to change that myself.

**Comment by Mikeo73** [comment]
Since there is no injection I agree with the update to remove the injection and later selling of the exp eggs and actions, however was it necessary to remove the equip. I.e. Would it be possible to keep some sort of equip exp egg and action (however many a person may already have available in their inventory) and then unequip them at the end. This can be done manually but it was nice that the program did it for you. 

Also will this no longer awaken units? I'm not sure if the past version was injecting awakening items or not.

**Comment by drchia** [comment]
It will no longer awaken units as it cannot inject the mats to do so. With the previous version, after injections were patched, it would still awaken units if you had the mats to do so in your inventory. If you didn't, it would crash. Now, it does not attempt at all.

The ability to equip the egg/action cannot be included because not everyone will have them. If it attempts to do so and fails, the tool crashes. The only reason the tool did it to begin with is due to the injection which guarantees the player would have the items/materia.

**Comment by Mikeo73** [comment]
It surprises me that it is impossible to do a check for actions and eggs. Get the number of them and then equip that available amount.

**Comment by drchia** [comment]
I doubt it's impossible, it's just more work for not a lot of return.

**Comment by Mikeo73** [comment]
Fair enough. I would still be interested in that adjustment. So if someone wants to add that feature I would find it helpful.

**Comment by drchia** [comment]
Oh sure, it would be nice to have, but I'm not about to pester the devs with requests like that. This forum is for features that are not working as intended, or bugs. The current rank up button breaks on rank 150 and ONLY rank 150 (since it doesn't say "if rank >=150", just "if rank=150") due to the recent cap change. Asking for added features is something completely different I think.

**Comment by TetoraK** [comment]
If you know the basics, simply compare old and new function and write your own.

Else another way: equip your units manually and use leveling funciton afterward.

**Comment by Mikeo73** [comment]
I made a suggestion about how to make the program better. I am not pestering anyone to do anything. If someone decides to add it great. If not they don't. But that is the intention of Github. People work together to make something better. Sorry if I put an question in an issues tab titled leveling tool questions.

**Comment by WithinGDR** [comment]
@drchia 

? You don't need to be a programmer
https://www.visualstudio.com/thank-you-downloading-visual-studio/?sku=Community&rel=15.
Download and install
https://github.com/shalzuth/BraveHaxvius/archive/master.zip
Download and extract

1. Double click BraveHaxvius.sln
2. On the right expand Client, right click MainWindow.cs and click View Code
3. Ctrl+F "150" replace that with "200" then Ctrl+S and Ctrl+Shift+B. Done.

@Mikeo73 

Just comment out the injection and selling of eggs then

**Comment by drchia** [comment]
@WithinGDR Thanks for the walkthrough!

@Mikeo73 I apologize if that came off the wrong way. No offense intended.
